keystone_api/apps/allocations/tasks.py

```diff
@@ -0,0 +1,184 @@
+"""Schedule tasks executed in parallel by Celery."""
+
+from datetime import date
+import logging
+import re
+from shlex import split
+from subprocess import PIPE, Popen
+from typing import Collection, List
+
+from celery import shared_task
+from django.db.models import Sum, ObjectDoesNotExist
+
+from apps.allocations.models import Allocation, Cluster
+from apps.users.models import ResearchGroup
+
+log = logging.getLogger(__name__)
+
+
+@shared_task()
+def update_limits() -> None:
+    """Adjust per Slurm account usage limits on all enabled clusters"""
+    log.info(f"{date.today()} - BEGIN update_limits")
+
+    for cluster in Cluster.objects.filter(enabled=True).all():
+        log.debug(f"Updating limits for cluster {cluster.name}")
+        update_limits_for_cluster(cluster)
+        log.debug(f"Updating limits for cluster {cluster.name} DONE")
+
+
+@shared_task()
+def update_limits_for_cluster(cluster: Cluster) -> None:
+    """Update the usage limits of each account on a given cluster, excluding the root account"""
+
+    for account_name in get_accounts_on_cluster(cluster.name):
+        if account_name in ['root']:
+            continue
+        log.debug(f"Updating limits for account {account_name}")
+        update_limit_for_account(account_name, cluster.name)
+        log.debug(f"Updating limits for account {account_name} DONE")
+
+
+@shared_task()
+def update_limit_for_account(account_name: str, cluster: Cluster) -> None:
+    """Update the usage limits for an individual Slurm account"""
+
+    # Check that the Slurm account has an entry in the keystone database
+    try:
+        account = ResearchGroup.objects.get(name=account_name)
+    except ObjectDoesNotExist:
+        #  Set the usage limit to zero (lock on this cluster) and continue
+        log.warning(f"No existing ResearchGroup for account {account_name}, locking {account_name} on {cluster.name}")
+        # TODO: This may create historical usage that keystone doesnt know about if raw usage is not reset
```

Comment:
Do not include TODO comments in merged code. Move them to the PR discussion so we can either address them or raise them as issues for further followup.